### DISCUSSION TITLE
Installable Modules

### DISCUSSION THREAD

**Discussion by eadlam** [cue-lang/cue#3494] [General]
I'm delivering an SDK to end users which includes a lot of cue modules. This currently works _ok_ by copying `cue.mod/pkg` into new projects. It would be better if they could be installed into a common system directory. It is not ideal for these to be fetched from a registry or to make the user run a local registry. Will the old style modules continue to be supported and has there been any more discussion around file system module management?

**Comment by myitcv** [comment]
@eadlam - great question. @rogpeppe and I were just discussing your situation. There are lots of possibilities/options, it really rather depends on what it is you are trying to do. As such, jumping onto a call is likely going be the most efficient way of helping improve our understanding, but also sharing with you the option space. We can then do a short write-up in this discussion for the benefit of others following along.

Please feel free to drop me an email paul@cue.works to get something scheduled if that works for you.

**Comment by the-nurk** [comment]
https://github.com/cue-lang/cue/issues/1013#issuecomment-3424493057

this is the pattern i was trying too. installable, copy-on-write modules directly out of the registry into a new project under that project's naming scheme.
